llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -805,6 +808,10 @@ void PassBuilder::addPGOInstrPasses(ModulePassManager &MPM,
   // Do counter promotion at Level greater than O0.
   Options.DoCounterPromotion = true;
   Options.UseBFIInPromotion = IsCS;
+  if (EnableSampledInstr) {
+    Options.Sampling = true;
+    Options.DoCounterPromotion = false;
```

Comment:
Can you add a comment on why counter promotion is turned off?

Reply:
The reason is mentioned in InstrProfling.C:400.